created PlaneBuilder

yurwing opened the pull request.

yurwing committed:
created PlaneBuilder

boroda4436 commented on src/main/java/core/basesyntax/Plane.java:
```diff
@@ -1,5 +1,80 @@
+        }
+
+        public Plane build() {
+            if (this.airframe == false
```
ok, let's add one more constructor that will initialize the required fields

yurwing committed:
added constructor with the required fields
created PlaneBuilder

boroda4436 commented on src/main/java/core/basesyntax/Plane.java:
```diff
@@ -1,5 +1,86 @@
+    private String model;
+    private String color;
+    private double maxSpeed;
+    private boolean propeller;
```
```suggestion
    private boolean hasPropeller;
```

boroda4436 commented on src/main/java/core/basesyntax/Plane.java:
```diff
@@ -1,5 +1,86 @@
+    private String color;
+    private double maxSpeed;
+    private boolean propeller;
+    private boolean airframe;
```
```suggestion
    private boolean hasAirframe;
```

boroda4436 commented on src/main/java/core/basesyntax/Plane.java:
```diff
@@ -1,5 +1,86 @@
+    private double maxSpeed;
+    private boolean propeller;
+    private boolean airframe;
+    private boolean wings;
```
```suggestion
    private boolean hasWings;
```
https://wiki.eclipse.org/Recommenders/CodingConventions#:~:text=is%20prefix%20should%20be%20used%20for%20boolean%20variables%20and%20methods.,-isSet%2C%20isVisible%2C%20isFinished&text=This%20is%20the%20naming%20convention,names%20like%20status%20or%20flag.

yurwing committed:
boolean names changed
added constructor with the required fields
created PlaneBuilder

yurwing committed:
boolean names changed 2.0
added constructor with the required fields
created PlaneBuilder

yurwing committed:
boolean names changed 3.0
added constructor with the required fields
created PlaneBuilder

KseniiaMakarova commented on src/main/java/core/basesyntax/BuilderTestApp.java:
```diff
@@ -1,7 +1,16 @@
+        System.out.println(plane.toString());
+        Plane plane1 = new Plane.PlaneBuilder(false, true, true)
+                .setColor("green")
+                .build();
```
if this line causes RuntimeException, let's remove it from `main` method

KseniiaMakarova commented on src/main/java/core/basesyntax/BuilderTestApp.java:
```diff
@@ -1,7 +1,16 @@
+                .setMaxSpeed(1000)
+                .setColor("white")
+                .build();
+        System.out.println(plane.toString());
```
```suggestion
        System.out.println(plane);
```
you can write like this as well

KseniiaMakarova commented on src/main/java/core/basesyntax/Plane.java:
```diff
@@ -1,5 +1,85 @@
+
+    @Override
+    public String toString() {
+        return new StringBuilder().append("Builder{")
```
let's not use StringBuilder here

yurwing committed:
remove StringBuilder and the constructor call in "main" method
boolean names changed 3.0
added constructor with the required fields
created PlaneBuilder
